src/runtime-rs/crates/hypervisor/src/ch/ch_api.rs

```diff
@@ -0,0 +1,301 @@
+use anyhow::{anyhow, Context, Result};
+
+use crate::ch::utils::{get_sandbox_path, get_vsock_path};
+use ch_config::{DeviceConfig, FsConfig, PmemConfig};
+use std::fmt::Display;
+use std::net::Ipv4Addr;
+use std::os::unix::net::UnixStream;
+use std::path::PathBuf;
+use tokio::task;
+
+use api_client::simple_api_full_command_and_response;
+
+use ch_config::net_util::MAC_ADDR_LEN;
+use ch_config::{
+    ConsoleConfig, ConsoleOutputMode, CpuTopology, CpusConfig, MacAddr, MemoryConfig, NetConfig,
+    PayloadConfig, RngConfig, VmConfig, VsockConfig,
+};
+
+pub async fn cloud_hypervisor_vmm_ping(mut socket: UnixStream) -> Result<()> {
+    task::spawn_blocking(move || -> Result<()> {
+        let response = simple_api_full_command_and_response(&mut socket, "GET", "vmm.ping", None)
+            .map_err(|e| anyhow!(e))?;
+
+        if response.is_some() {
+            warn!(sl!(), "Cloud Hypervisor ping returned: {:?}", response);
```

Comment:
Does this indicates that the `vmm.ping`'s response is not expected? Otherwise, I would say `info` or `debug` level make more sense.

Reply:
All these similar comments have been resolved by allowing the caller to handle the response.